content/vendor/guides/persistent-datastores.md

```diff
@@ -0,0 +1,940 @@
+
+### Adding the Secret and StatefulSet
+
+Firse, we'll create a secret to store the root password for our embedded postgres instance. 
+
+```yaml
+# postgres-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: postgres
+data:
+  DB_PASSWORD: '{{repl ConfigOption "embedded_postgres_password" | Base64Encode }}'
+```
+
+Next, create a new YAML file in your `manifests` directory with the following contents. Note the use of `kots.io/when` to only conditionally include this based on end-user inputs.
+
+```yaml
+# postgres-statefulset.yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: postgres
+  labels:
+    app: pg-provider
+  annotations:
+    kots.io/when: '{{repl ConfigOptionEquals "postgres_type" "embedded_postgres" }}'
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: pg-provider
+  serviceName: postgres
+  template:
+    metadata:
+      labels:
+        app: pg-provider
+    spec:
+      containers:
+      - env:
+        - name: PGDATA
+          value: /var/lib/postgresql/data/pgdata
+        # create a db called "postgres"
+        - name: POSTGRES_DB
+          value: postgres
+        # create admin user with name "postgres"
+        - name: POSTGRES_USER
+          value: postgres
+        # use admin password from secret
+        - name: POSTGRES_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              key: DB_PASSWORD
+              name: postgres-secret
+        image: postgres:10
+        name: postgres
+        volumeMounts:
+        - mountPath: /var/lib/postgresql/data
+          name: pgdata
+      volumes:
+      - name: pgdata
+        persistentVolumeClaim:
+          claimName: pgdata
+  volumeClaimTemplates:
+  - metadata:
+      name: pgdata
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Finally, lets add a Service object so we can route traffic to our postgres instance, again using `kots.io/when` to conditionally include this resource.
+
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres
+  labels:
+    app: pg-provider
+  annotations:
+    kots.io/when: '{{repl ConfigOptionEquals "postgres_type" "embedded_postgres" }}'
+spec:
+  ports:
+  - port: 5432
+  selector:
+    app: pg-provider
+  type: ClusterIP
+```
+
+### Validating the embedded Database
+
+Once you've added the these resources, you can push a new release and update in kotsadm. You should see the following in the deployment logs:
+
+![Embedded PG Deployed](/images/guides/kots/embedded-pg-deployed.png)
+
+We should now see an instance of postgres running in our namespace as well. The consumer may still be crashlooping, but we can see the error is different now:
+
+```text
+$ kubectl logs -l app=pg-consumer 
+psql: FATAL:  password authentication failed for user "postgres"
+```
+
+This is because we still need to deliver the generated password to our workload pod. In `pg-consumer.yaml`, we'll remove this section
```

Comment:
So either we show how to make postgres use the generated password (instruct user to delete 'postgres-0') or we make it work with 'postgres' password. As the steps are now, it fails with an auth error.

Reply:
Can you explain this a bit more -- 

We do this

> we show how to make postgres use the generated password

But I don't understand what case we'd need to  

>  delete the associated pvc

If the first time things are run with "embedded postgres", and the password is never changed, in which case would we need to reset the db state?

Reply:
I'm 80% sure that the above issue was a red herring because of the DB_USER problem (where a space was causing the pg-consumer to fail). So when I tried install to new cluster, it failed out of the box. I was thinking that it was an order of operations failure, but it could just have been that issue. 

Safe to disregard, I think.